src/v/datalake/tests/parquet_writer_test.cc

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "datalake/arrow_translator.h"
+#include "datalake/parquet_writer.h"
+#include "datalake/tests/test_data.h"
+#include "iceberg/tests/value_generator.h"
+#include "utils/file_io.h"
+
+#include <arrow/array.h>
+#include <arrow/io/interfaces.h>
+#include <arrow/io/memory.h>
+#include <arrow/table.h>
+#include <arrow/type_fwd.h>
+#include <gtest/gtest.h>
+#include <parquet/arrow/reader.h>
+#include <parquet/type_fwd.h>
+
+TEST(ParquetWriter, DoesNothing) {
+    datalake::arrow_translator schema_translator(
+      test_schema(iceberg::field_required::no));
+
+    iobuf full_result;
+
+    for (int i = 0; i < 5; i++) {
+        auto data = iceberg::tests::make_value(
+          iceberg::tests::value_spec{
+            .forced_fixed_val = iobuf::from("Hello world")},
+          test_schema(iceberg::field_required::no));
+        schema_translator.add_data(std::move(data));
+    }
+
+    std::shared_ptr<arrow::Array> result = schema_translator.take_chunk();
+    ASSERT_NE(result, nullptr);
+
+    datalake::arrow_to_iobuf writer(*schema_translator.build_arrow_schema());
+
+    // The first write is a special case because it is 4 bytes longer.
+    writer.add_arrow_array(result);
+    iobuf serialized = writer.take_iobuf();
+    EXPECT_NEAR(serialized.size_bytes(), 3300, 200);
+    full_result.append_fragments(std::move(serialized));
+
+    for (int i = 0; i < 10; i++) {
+        writer.add_arrow_array(result);
+        serialized = writer.take_iobuf();
+        // Sizes are not consistent between writes, but should be about right.
+        EXPECT_NEAR(serialized.size_bytes(), 3300, 200);
+        full_result.append_fragments(std::move(serialized));
+    }
+
+    // The last write is also long. This is probably Parquet footer information.
+    serialized = writer.close_and_take_iobuf();
+    EXPECT_NEAR(serialized.size_bytes(), 22881, 1000);
+    full_result.append_fragments(std::move(serialized));
+
+    EXPECT_NEAR(full_result.size_bytes(), 60000, 1000);
+
+    // Check that the data is a valid parquet file. Convert the iobuf to a
+    // single buffer then import that into an arrow::io::BufferReader
+    auto vbegin = iobuf::byte_iterator(
+      full_result.cbegin(), full_result.cend());
+    auto vend = iobuf::byte_iterator(full_result.cend(), full_result.cend());
+    std::string full_result_string;
+    // Byte iterators don't work with the string constructor.
+    while (vbegin != vend) {
+        full_result_string += *vbegin;
+        ++vbegin;
+    }
```

Comment:
does this work:

auto b = iobuf_to_bytes()
std::string(b.c_str(), b.size())

or add a helper for this case somewhere like bytes/string.h you can model it after iobuf_to_bytes in bytes.h.